software/tracksight/frontend/lib/types/Signal.ts

```diff
@@ -0,0 +1,18 @@
+type SignalMetadata = {
```

Comment:
i would suggest modelling the data (in a similar way?) to as how it's doing it in the backend, namely I think the only signal types that we have are numerical, enum and alert (boolean)